### PR TITLE
add well-known config for OIDC

### PR DESCRIPTION
**Pull request by techknowlogick**

### CODE REVIEW

File: templates/user/auth/oidc_wellknown.tmpl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+	"issuer": "{{AppUrl | JSEscape | Safe}}",
+	"authorization_endpoint": "{{AppUrl | JSEscape | Safe}}login/oauth/authorize",
+	"token_endpoint": "{{AppUrl | JSEscape | Safe}}login/oauth/access_token",
+	"userinfo_endpoint": "{{AppUrl | JSEscape | Safe}}login/oauth/userinfo",
```
Comment (silverwind) [inline templates/user/auth/oidc_wellknown.tmpl]: Is `AppUrl` guaranteed to contain a trailing slash?

**Comment by silverwind** [comment]
The template file is tab-indented but I think we generally space-indent json templates. Could you convert and add an entry to `.editoconfig` for it, like was done here?

https://github.com/go-gitea/gitea/pull/15289/files#diff-0947e2727d6bad8cd0ac4122f5314bb5b04e337393075bc4b5ef143b17fcbd5bR18-R19

Maybe we should move json templates to `templates/json` and match that directory.

File: templates/user/auth/oidc_wellknown.tmpl (hunk 1 above)
Comment (techknowlogick) [reply templates/user/auth/oidc_wellknown.tmpl]: yes, look at modules/setting/setting.go Where AppURL is `AppURL = strings.TrimRight(AppURL, "/") + "/"` (note the different capitalization, this var is passed to the template using the expected capitalization I've used in the template)

**Comment by techknowlogick** [comment]
@silverwind updated per your feedback. Thanks :)

**Review by silverwind — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

File: templates/user/auth/oidc_wellknown.tmpl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "issuer": "{{AppUrl | JSEscape | Safe}}",
+    "authorization_endpoint": "{{AppUrl | JSEscape | Safe}}login/oauth/authorize",
+    "token_endpoint": "{{AppUrl | JSEscape | Safe}}login/oauth/access_token",
+    "userinfo_endpoint": "{{AppUrl | JSEscape | Safe}}login/oauth/userinfo",
```
Comment (6543) [inline templates/user/auth/oidc_wellknown.tmpl]: I would say blocked by #14938 ?

Comment (silverwind) [reply templates/user/auth/oidc_wellknown.tmpl]: Would say remove it here, land this and then suggest the addition in that PR.

Comment (techknowlogick) [reply templates/user/auth/oidc_wellknown.tmpl]: updated per silverwind suggestion.

**Pull request merged**